controllers/users/checkAllowAnonDM.js

```diff
@@ -0,0 +1,21 @@
+const {User} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  let {userId} = req.params;
+  let user = await User.findOne({where: {user_id: userId, is_anonymous: false}});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `findOne` function is an asynchronous operation that can throw an error if the database connection fails or if there's an issue with the query. It's best practice to wrap this in a try-catch block to handle potential errors.
```diff
-  let user = await User.findOne({where: {user_id: userId, is_anonymous: false}});
+  let user;
+  try {
+    user = await User.findOne({where: {user_id: userId, is_anonymous: false}});
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({
+      status: 'error',
+      message: 'Database Error'
+    });
+  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->